state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -143,16 +146,32 @@ impl OnCheckComplete<<Runtime as Chainflip>::ValidatorId> for OnCheckCompleteHoo
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, TypeInfo)]
 pub struct TransactionSuccessDetails {
 	pub tx_fee: u64,
+	// It is possible for a contract call to be reverted due to contract's internal error.
+	// This field is set to `true` if the contract call executed successfully without error.
+	pub contract_call_successful: bool,
 }
 
 pub struct SolanaEgressWitnessingHook;
 
 impl OnEgressSuccess<SolSignature, TransactionSuccessDetails> for SolanaEgressWitnessingHook {
 	fn on_egress_success(
 		signature: SolSignature,
-		TransactionSuccessDetails { tx_fee }: TransactionSuccessDetails,
+		TransactionSuccessDetails { tx_fee, contract_call_successful }: TransactionSuccessDetails,
 	) {
 		use cf_traits::KeyProvider;
+		if !contract_call_successful {
+			// On CCM failure, we need to refund the user using their fallback info.
+			if let Some((broadcast_id, ccm_tx)) =
+				SolanaBroadcaster::pending_api_call_from_out_id(signature)
+			{
+				// Only Ccm calls support fallback.
+				if let SolanaTransactionType::CcmTransfer { fallback } = ccm_tx.call_type {
+					SolanaIngressEgress::do_ccm_fallback(broadcast_id, fallback);
```

Comment:
Should we emit some event for the frontend to detect when a ccm transaction has reverted or we can rely on them understanding that when they see a `CcmFallbackScheduled` event with a broadcast Id it means that the CCM has reverted even if the egress is succesful? I presume that they use egress success as a proxy for transaction has succeeded. @acdibble

Reply:
Yes we have an event emitted when the fallback is triggered: 
Event::<T, I>::CcmFallbackScheduled {
	broadcast_id,
	egress_details,
}